From 8a5fe0c06f48b202a5f1d3dd49f7ed7ed92d64ae Mon Sep 17 00:00:00 2001
From: Timo Sirainen <timo.sirainen@dovecot.fi>
Date: Fri, 9 Jun 2017 14:31:15 +0300
Subject: [PATCH] lib-storage: Fix setting the correct cache record corrupted

It was mixing UIDs and sequences, so a wrong mail could have been set
corrupted or it could have crashed with:

Panic: file mail-index-transaction-update.c: line 1018 (mail_index_update_ext): assertion failed: (seq > 0 && (seq <= mail_index_view_get_messages_count(t->view) || seq <= t->last_new_seq))
---
 src/lib-storage/mail-storage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib-storage/mail-storage.c b/src/lib-storage/mail-storage.c
index 067a65bc1..2405161e0 100644
--- src/lib-storage/mail-storage.c
+++ src/lib-storage/mail-storage.c
@@ -2754,7 +2754,7 @@ void mail_set_mail_cache_corrupted(struct mail *mail, const char *fmt, ...)
 	va_start(va, fmt);
 
 	T_BEGIN {
-		mail_cache_set_seq_corrupted_reason(cache_view, mail->uid,
+		mail_cache_set_seq_corrupted_reason(cache_view, mail->seq,
 			t_strdup_printf("UID %u: %s",
 					mail->uid,
 					t_strdup_vprintf(fmt, va)));
